### ISSUE TITLE
Unable to hover over certain error popups

### ISSUE THREAD

**Issue by max-petrenko** [zed-industries/zed#13309]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When trying to hover onto the Typescript error popup that is attached to the `satisfies` keyword, it disappears even on the slightest mouse movement, making it impossible to click on in to unfold in inline (or do anything else with it). It acts like that regardless of the window spawn position (top or bottom of the keyword).

To the contrary, if the same error TS error occurs and is attached to the variable, it can be hovered (see the video).

### Environment

Zed: v0.140.5 (Zed)
OS: macOS 14.4.0
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/zed/assets/101061124/1e93bd89-2fb0-449e-a336-8fb90a477a48

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

Unapplicable

**Comment by notcodev** [comment]
I agree, this is a very annoying problem. I think it can be solved by adding a delay before hiding popup.
